### ISSUE TITLE
[Bug]: Valuables dissappear from refitable cargo list for some of newgrfs

### ISSUE THREAD

**Issue by tlpwka** [JGRennison/OpenTTD-patches#773]
### Version of OpenTTD

0.63.0

### Expected result

As in 0.62.0, vehicles should have option to refit to valuables
![1](https://github.com/user-attachments/assets/3e563187-068c-4b83-8300-5898aecc2050)

### Actual result

Cannot be refitted in 0.63.0
![2](https://github.com/user-attachments/assets/2914712b-ac31-444e-a143-469856d458fa)
Impacted at least:

- Bobs random British vehicles 
- Russian trucks v0.2
- Hungarian truck set v2.1

not impacted:
- eGRVTS  v2.0
- Horse carts road vehicles set 1.2

### Steps to reproduce

[0.62.0.sav.zip](https://github.com/user-attachments/files/17693195/0.62.0.sav.zip)
[0.63.0.sav.zip](https://github.com/user-attachments/files/17693196/0.63.0.sav.zip)

**Comment by JGRennison** [comment]
This GRF/climate configuration doesn't work for valuables in current vanilla OpenTTD either.
These vehicle sets are very old and don't interact particularly well with other NewGRFs.

Most likely the specific change is: https://github.com/OpenTTD/OpenTTD/pull/12646

**Comment by tlpwka** [comment]
that is sad.
Do you think the impacted grfs will be hard to fix by myself (i did not checked other truck set if they have any issues)? I can't find source files for this particular sets, so there is only grfcodec option to decode them...

**Comment by PeterN** [comment]
Yeah, OpenTTD/OpenTTD#12646 fixes things for old vehicle sets that assumed the default cargo types are available. With non-default cargo types it was already mostly luck what happened, and was possibly not as intended -- without access to the original documentation or source there is no indication of what is expected.

GRF Overrides should allow the original cargo type properties and cargo classes to be set for these older GRFs, and OpenTTD/OpenTTD#/12879 might help by letting other GRFs install an appropriate cargo translation table onto them, if fixing them natively is out of the question.

**Comment by PeterN** [comment]
I think https://github.com/OpenTTD/OpenTTD/pull/13131 should fix most of these, although I've only tested with US Trains Set.
